internal/controller/volumereplicationgroup_controller.go

```diff
@@ -1167,12 +1167,37 @@ func (v *VRGInstance) removeFinalizer(finalizer string) error {
 	return nil
 }
 
+func (v *VRGInstance) resetKubeObjectsCaptureStatusIfRequired() {
+	if v.kubeObjectProtectionDisabled("capture") {
+		// set the in-memory condition to nil to indicate that we don't need
+		// kube objects protected for this vrg
+		v.kubeObjectsCaptureStatusNil()
+
+		return
+	}
+
+	if v.instance.Spec.PrepareForFinalSync || v.instance.Spec.RunFinalSync {
+		// if we are in final sync, then we don't reset the condition
+		return
+	}
+
+	// clusterDataProtected looks at the v.kubeObjectsProtected condition
+	// to determine if the cluster data is protected. If it is nil, then it is
+	// considered as success. So we should set it as false here and set it as
+	// true if protection is not required or if protection is successful.
+	v.kubeObjectsCaptureStatusFalse("KubeObjectsCaptureNotStarted", "Kube objects capture has not started")
+}
+
 // processAsPrimary reconciles the current instance of VRG as primary
+//
+//nolint:cyclop,funlen,gocognit
 func (v *VRGInstance) processAsPrimary() ctrl.Result {
 	v.log.Info("Entering processing VolumeReplicationGroup as Primary")
 
 	defer v.log.Info("Exiting processing VolumeReplicationGroup")
 
+	v.resetKubeObjectsCaptureStatusIfRequired()
```

Comment:
Why a reset? Isn't the last saved condition all you need? The reason why I am asking is because of line 1222. If we get to that line, then the condition will be reset even if we have successfully captured in the previous reconciliation.

Reply:
There are two reasons for this
1. the init value of this condition when kube object protection is disabled is nil and when it is enabled is false
2. we need to treat the condition as false at every reconcile start as false because we don't know if the capture interval has elapsed since the last capture.

The other way to fix this would be to inspect the last capture time and then only reset it if the capture interval has passed. That would be a lot of calculation in the init process is how I see it.